storage/src/main/java/org/apache/uniffle/storage/handler/impl/ComposedClientReadHandler.java

```diff
@@ -127,7 +127,7 @@ public ShuffleDataResult readShuffleData() {
           return null;
       }
     } catch (Exception e) {
-      LOG.error("Failed to read shuffle data from " + getCurrentHandlerName() + " handler", e);
+      throw new RssException("Failed to read shuffle data from " + getCurrentHandlerName() + " handler", e);
```

Comment:
Why do we need to throw this exception?

Reply:
> It should fast fail on network failure when reading shuffle data from memory/localfile/hdfs in ComposedClientReadHandler. If not, it will throw inconsistent blockIds and make users confused.

Reply:
Make sense.

Reply:
But if you read data from memory, and this data is writing to the localfile, the data is incomplete, when we read the incomplete data, it will throw an exception. The exception prevent us reading the data from HDFS.

Reply:
I think it wont happen that client reads the incomplete blocks from server's memory. 

I think we should handle all the retriable exceptions and then fail fast for other exceptions.

Reply:
> I think it wont happen that client reads the incomplete blocks from server's memory.
> 
> I think we should handle all the retriable exceptions and then fail fast for other exceptions.

Is it possible to read incomplete local file data?

Reply:
It wont. Refer https://github.com/apache/incubator-uniffle/blob/f1cb43fd3928daf845f8d5a6495e51f40a98ac10/common/src/main/java/org/apache/uniffle/common/util/RssUtils.java#L204-L232

Reply:
OK.